include/circt/Dialect/FIRRTL/Passes.h

```diff
@@ -47,6 +47,8 @@ std::unique_ptr<mlir::Pass> createGrandCentralPass();
 
 std::unique_ptr<mlir::Pass> createGrandCentralTapsPass();
 
+std::unique_ptr<mlir::Pass> createDetectCombLoopsPass();
```

Comment:
General comment, please name this "DetectCombCycles" instead of "loops".  "Loops" means something pretty specific in the compiler world (at least my brain is programmed to think so).  I think that "cycles" captures the more general graph property that you're checking.